app/src/main/java/bg/zahov/app/ui/workout/WorkoutViewModelNew.kt

```diff
@@ -0,0 +1,905 @@
+     *
+     * This function organizes the workout data into exercises, calculates the workout volume, and optionally updates
+     * the exercise data in the repository. It also compares the best sets of current exercises with template exercises
+     * to determine if a new personal record (PR) is achieved.
+     *
+     * @param entries A list of `WorkoutEntry` objects representing exercises and sets in the workout.
+     * @param updateExercises A boolean flag indicating whether to update the exercises in the repository.
+     * If we save it to realm there is no need to update them in the firestore repository
+     * @return A `Triple` containing:
+     *  - A `List` of `Exercise` objects reflecting the processed workout.
+     *  - An `Int` representing the count of PRs achieved during the workout.
+     *  - A `Double` representing the total workout volume (sum of weight * repetitions for all sets).
+     *
+     * ## Details:
+     * 1. **Extract and Calculate**:
+     *    - Calls `getExercisesFiltered` to extract exercises and calculate the total volume.
+     * 2. **PR Comparison** (if `updateExercises` is `true`):
+     *    - Compares the best set of each current exercise to its corresponding template exercise (if any).
+     *    - Updates the `Exercise.bestSet` to retain the higher-performing set.
+     *    - Increments the PR count (`prs`) if the current best set outperforms the template best set.
+     *    - Marks template exercises as such (`isTemplate = true`).
+     * 3. **Repository Update**:
+     *    - Updates the exercise repository with the modified exercises.
+     */
+    private suspend fun getExerciseArrayAndPRs(
+        entries: List<WorkoutEntry>,
+        updateExercises: Boolean = true,
+    ): Triple<List<Exercise>, Int, Double> {
+        val (exercises, volume) = getExercisesFiltered(entries)
+        var prs = 0
+
+
+        if (updateExercises) {
+
+            templateExercises.forEach { previous ->
+                val curr = exercises[previous.name]
+                curr?.let { current ->
+                    val currentBestSetResult = (current.bestSet.firstMetric
+                        ?: 0.0) * (current.bestSet.secondMetric ?: 0)
+                    val previousBestSetResult =
+                        (previous.bestSet.firstMetric
+                            ?: 0.0) * (previous.bestSet.secondMetric
+                            ?: 0)
+
+                    if (currentBestSetResult <= previousBestSetResult) {
+                        exercises[previous.name]?.copy(bestSet = previous.bestSet)
+                     } else {
+                        prs++ // Increment PR count if the current best set surpasses the template best set.
+                    }
+
+                    exercises[previous.name]?.copy(isTemplate = true)
+                }
+            }
+
+            repo.updateExercises(exercises.values.toList())
+        }
+        return Triple(exercises.values.toList(), prs, volume)
```

Comment:
Use data class, instead of Triple it will be more clear and expressive

Reply:
🫡